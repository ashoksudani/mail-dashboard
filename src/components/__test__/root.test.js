import React from 'react';
import Root from '../root';

test('renders with expected list of components', () => {
  const rootWrapper = mount(<Root />);

  expect(rootWrapper).toMatchSnapshot();

  //login form
  expect(rootWrapper.find('form').length).toEqual(1);
  expect(rootWrapper.find('input[name="emailId"]').length).toEqual(1);
  expect(rootWrapper.find('input[name="password"]').length).toEqual(1);
});
