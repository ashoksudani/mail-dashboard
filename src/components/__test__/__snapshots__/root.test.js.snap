// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with expected list of components 1`] = `
<div
  className="login-form"
>
  <style>
    
        body &gt; div,
        body &gt; div &gt; div,
        body &gt; div &gt; div &gt; div.login-form {
          height: 100%;
        }
      
  </style>
  <div
    className="ui centered center aligned middle aligned grid"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <h2
        className="ui teal center aligned header"
      >
        <div
          className="content"
        >
          Log-in to Mailbox
        </div>
      </h2>
      <form
        className="ui large form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="emailId"
              onChange={[Function]}
              placeholder="E-mail address"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
            />
          </div>
        </div>
        <button
          className="ui teal large fluid button"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
